### PR TITLE
Handle array for `with` option

### LINKED ISSUE #7: Using "with" Option to Render an Array
Currently, this is not available:

```crystal
class SomeRenderer < Crinder::Base(Some)
  field others, with: OtherRenderer # typeof(others) # => Array(Other)
end
```

### PR DESCRIPTION
**Pull request by alex-lairan**
The `with` option should handle Arrays.

Fix : https://github.com/c910335/crinder/issues/7

### CODE REVIEW

File: src/crinder.cr [hunk 1]
```diff
@@ -209,6 +205,14 @@ class Crinder::Base(T)
       end
     end
```
Comment (c910335) [inline src/crinder.cr:200]: Should merge these two `.render` above.

```crystal
def self.render(object : T | Array(T)) : String
```

File: spec/crinder_spec.cr [hunk 2]
```diff
@@ -47,6 +47,18 @@ class NilableTodoRenderer < TodoRenderer
   remove updated
 end
 
+record TodoSubtasked, name : String, subtasks : Array(Subtask)
+record Subtask, name : String
```
Comment (c910335) [inline spec/crinder_spec.cr:51]: I prefer moving these two `record`s to L4, under `record Todo`.

**Review by c910335 — CHANGES_REQUESTED** [review]
Sorry for late response.

The documentation of `with` option (`src/crinder.cr` L86) should also be fixed.

File: spec/crinder_spec.cr (hunk 2 above)
Comment (alex-lairan) [reply spec/crinder_spec.cr:51]: I think fixtures should be simple and focused to what's important.

File: src/crinder.cr (hunk 1 above)
Comment (alex-lairan) [reply src/crinder.cr:200]: Done :+1:

**Comment by alex-lairan** [comment]
Hi !

Thanks for the review.

About this documentation :

```   
# - **with**: a renderer for this field. This field will be filtered by `value` before passing to it. It is not necessary to be a subclass of `Crinder::Base`, but it must have the class method `render(object : T, json : JSON::Builder)` where `T` is the original type of this field.
```

I need to change `render(object : T, json : JSON::Builder)` to `render(object : T | Array(T), json : JSON::Builder)` ?

**Comment by c910335** [comment]
@alex-lairan 

> I need to change render(object : T, json : JSON::Builder) to render(object : T | Array(T), json : JSON::Builder) ?

Yes

**Comment by alex-lairan** [comment]
@c910335 I've finished the documentation :)

**Review by c910335 — APPROVED** [review]

**Pull request merged**